examples/knative/templates/server.ps1

```diff
@@ -1,141 +1,156 @@
-
-      $cloudEvent = $null
-
-      try {
-         $listener.Start()
-         $context = $listener.GetContext()
-
-         try {
-            # Read Input Stream
-            $buffer = New-Object 'byte[]' -ArgumentList 1024
-            $ms = New-Object 'IO.MemoryStream'
-            $read = 0
-            while (($read = $context.Request.InputStream.Read($buffer, 0, 1024)) -gt 0) {
-               $ms.Write($buffer, 0, $read);
-            }
-            $bodyData = $ms.ToArray()
-            $ms.Dispose()
-
-            # Read Headers
-            $headers = @{}
-            for($i =0; $i -lt $context.Request.Headers.Count; $i++) {
-               $headers[$context.Request.Headers.GetKey($i)] = $context.Request.Headers.GetValues($i)
+    param($url, $serverStopMessage)
+
+    Import-Module 'Microsoft.PowerShell.Utility'
+    Import-Module CloudEvents.Sdk
+
+    . ./handler.ps1
+
+    function Start-HttpCloudEventListener {
+        <#
+            .DESCRIPTION
+            Starts a HTTP Listener on specified Url
+        #>
+
+        [CmdletBinding()]
+        param(
+            [Parameter(
+                Mandatory = $true,
+                ValueFromPipeline = $false,
+                ValueFromPipelineByPropertyName = $false)]
+            [ValidateNotNull()]
+            [string]
+            $Url
+        )
+
+        $listener = New-Object -Type 'System.Net.HttpListener'
+        $listener.AuthenticationSchemes = [System.Net.AuthenticationSchemes]::Anonymous
+        $listener.Prefixes.Add($Url)
+
+        $cloudEvent = $null
+
+        try {
+            $listener.Start()
+
+            while ($true) {
+                $context = $listener.GetContext()
+
+                try {
+                    # Read Input Stream
+                    $buffer = New-Object 'byte[]' -ArgumentList 1024
+                    $ms = New-Object 'IO.MemoryStream'
+                    $read = 0
+                    while (($read = $context.Request.InputStream.Read($buffer, 0, 1024)) -gt 0) {
+                        $ms.Write($buffer, 0, $read);
+                    }
+                    $bodyData = $ms.ToArray()
+                    $ms.Dispose()
+
+                    # Read Headers
+                    $headers = @{}
+                    for ($i = 0; $i -lt $context.Request.Headers.Count; $i++) {
+                        $headers[$context.Request.Headers.GetKey($i)] = $context.Request.Headers.GetValues($i)
+                    }
+
+                    $cloudEvent = ConvertFrom-HttpMessage -Headers $headers -Body $bodyData
+
+                    if ([System.Text.Encoding]::UTF8.GetString($bodyData) -eq $serverStopMessage) {
+                        # Server Stop request
+                        $context.Response.StatusCode = [int]([System.Net.HttpStatusCode]::OK)
+                        $context.Response.Close();
+                        # Set function reault
+                        $true
+                        break
```

Comment:
I read this as break try/catch and continue with while loop which should stop then right? But shouldn't it be $false then instead to end the while loop?

Reply:
`break` breaks the loop. The sense of result is `true` break signal recevied. I'll add the result documentation in the fuction help

Reply:
Addressed with documentation